Source/WebKit/UIProcess/XR/openxr/PlatformXROpenXR.cpp

```diff
@@ -188,6 +280,109 @@ void OpenXRCoordinator::initializeDevice()
     collectViewConfigurations();
 }
 
+void OpenXRCoordinator::initializeGraphicsBinding()
+{
+    if (!m_platformDisplay)
+        m_platformDisplay = WebCore::PlatformDisplaySurfaceless::create();
+    if (!m_glContext)
+        m_glContext = WebCore::GLContext::createOffscreen(*m_platformDisplay);
+
+    m_graphicsBinding = createOpenXRStruct<XrGraphicsBindingEGLMNDX, XR_TYPE_GRAPHICS_BINDING_EGL_MNDX>();
+    m_graphicsBinding.display = m_platformDisplay->eglDisplay();
+    m_graphicsBinding.context = m_glContext->platformContext();
+    m_graphicsBinding.config = m_glContext->config();
+    m_graphicsBinding.getProcAddress = m_extensions->methods().getProcAddressFunc;
+}
+
+void OpenXRCoordinator::createSessionIfNeeded()
+{
+    ASSERT(RunLoop::isMain());
+    ASSERT(m_instance != XR_NULL_HANDLE);
+
+    if (m_session != XR_NULL_HANDLE)
+        return;
+
+#if defined(XR_USE_GRAPHICS_API_OPENGL_ES)
+    auto requirements = createOpenXRStruct<XrGraphicsRequirementsOpenGLESKHR, XR_TYPE_GRAPHICS_REQUIREMENTS_OPENGL_ES_KHR>();
+    CHECK_XRCMD(m_extensions->methods().xrGetOpenGLESGraphicsRequirementsKHR(m_instance, m_systemId, &requirements));
```

Comment:
We should check whether the function pointer returned by the `xrGetInstanceProcAddr()` from earlier returned a null pointer to avoid a crash here. In that case, given that at the moment we are only supporting OpenGL ES, leave the session as null and return early here.

```suggestion
    if (!m_extensions->methods().xrGetOpenGLESGraphicsRequirementsKHR) {
        LOG(XR, "Function xrGetOpenGLESGraphicsRequirementsKHR() is not available, cannot create session");
        return;
    }
    
    CHECK_XRCMD(m_extensions->methods().xrGetOpenGLESGraphicsRequirementsKHR(m_instance, m_systemId, &requirements));
```

Reply:
We'll eventually have some other methods that would be used in various parts of the code. Instead of the above suggestion I'll add the check to the OpenXRExtensions::loadMethods() so that we don't go on with the rest of the setup process